Add support for role mapping and scheduled tasks

lsimons opened the pull request:
I don't have nexus professional so I can't test that I didn't break nexus professional support, but I doubt I broke anything.

Tested against latest nexus only but from the API docs it seems likely most of these commands will work with older nexus installs, too.

There is a bit too much copy-pasting in lib/nexus_cli/tasks.rb for my taste, but I couldn't figure out a clean way to avoid it without making the code complex.

lsimons committed:
Adding role mapping support

lsimons committed:
Ignore intellij module file

lsimons committed:
Add scheduled tasks support

lsimons committed:
Fix some CLI docs for scheduled tasks support

lsimons committed:
Note scheduled tasks support in changelog

lsimons commented:
This is a dependency for https://github.com/RiotGamesCookbooks/nexus-cookbook/pull/107 .

lsimons committed:
Avoid sending null to nexus, which it doesn't like

lsimons committed:
Role mappings are case sensitive

lsimons committed:
Role mappings are case sensitive

lsimons committed:
Role mappings are case sensitive -- learning to type

KAllan357 commented on lib/nexus_cli/mixins/task_actions.rb:
```diff
@@ -0,0 +1,247 @@
+
+    def create_download_indexes_task(name, schedule='daily', enabled=true, start_date=nil, recurring_time=nil,
+                                     repository_id='all_repo')
+      params = common_task_params('DownloadIndexesTask', enabled, name, recurring_time, schedule, start_date,
```
I wonder if it might help organize this code a bit if all of the "task types" were pulled up at setup as constants at the top.

ex: `DOWNLOAD_INDEX_TASK = 'DownloadIndexesTask'`

KAllan357 commented on lib/nexus_cli/tasks.rb:
```diff
@@ -5,6 +5,7 @@ module NexusCli
   module Tasks
     def self.included(base)
       base.send :include, ::Thor::Actions
+      # noinspection RubyInstanceMethodNamingConvention
```
I see a few of these comments, but my Google searches have been coming up empty. Could you tell me what these do?

lsimons replied:
These are injected by rubymine (or intellij) to tell it to ignore style guide violations, like too long class or method names. I don't know if other analyzers support them. I guess I should take them out :)

KAllan357 commented on VERSION:
```diff
@@ -1 +1 @@
-4.1.0
\ No newline at end of file
+4.1.1
```
Could you revert this? I can take care of bumping the version.

lsimons replied:
will do!

KAllan357 commented:
@lsimons Sorry for the delay in looking this over. I think this is all really great - its nice to see some more functionality added in this gem. I left a few minor comments, but I'll give this another day or two and :+1:

lsimons commented:
key @KAllan357 no worries, busy is normal. I'll go update the pull req.

lsimons committed:
Small changes after code review.

* Revert version bump
* Remove intellij code inspection comments.
* Introduce constants for task types.

Note: the corresponding outstanding nexus cookbook changes depend on
the version bump, so I cannot easily integration-test this.
